src/bvals/bvals_base.cpp

```diff
@@ -51,9 +51,9 @@ NeighborIndexes BoundaryBase::ni[NMAX_NEIGHBORS];
 //                          int ibid, int itargetid, int ifi1=0, int ifi2=0)
 // \brief Set neighbor information
 
-void NeighborBlock::SetNeighbor(int irank, int ilevel, int igid, int ilid, int iox1,
-                                int iox2, int iox3, NeighborConnect itype, int ibid,
-                                int itargetid,
+void NeighborBlock::SetNeighbor(LogicalLocation inloc, int irank, int ilevel, int igid,
```

Comment:
Assuming here that `irank` is mpi rank? I'm not sure I like the idea of that information being exposed unless it has to be. What is that used for here? It's not needed for the space filling curve is it?

Reply:
IIRC, I you need it for building the boundary buffers. It doesn't enter into the Morton order.

Reply:
why isn't it in the older function if it's needed?

Reply:
I think it is in the older one...

Reply:
oh I'm just blind.